### PR TITLE
Node v8 compatibility

### LINKED ISSUE #1769: node server & upgrade broken with Node v8
## Error Log
<details>
<summary><code>node upgrade</code></summary>

```console
module.js:515
    throw err;
    ^

Error: Cannot find module 'internal/util/types'
    at Function.Module._resolveFilename (module.js:513:15)
    at Function.Module._load (module.js:463:25)
    at Module.require (module.js:556:17)
    at require (internal/module.js:11:18)
    at evalmachine.<anonymous>:31:26
    at Object.<anonymous> (D:\adapt_authoring\node_modules\bower\node_modules\graceful-fs\fs.js:11:1)
    at Module._compile (module.js:612:30)
    at Object.Module._extensions..js (module.js:623:10)
    at Module.load (module.js:531:32)
    at tryModuleLoad (module.js:494:12)
    at Function.Module._load (module.js:486:3)
    at Module.require (module.js:556:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (D:\adapt_authoring\node_modules\bower\node_modules\graceful-fs\graceful-fs.js:3:27)
    at Module._compile (module.js:612:30)
    at Object.Module._extensions..js (module.js:623:10)
```
</details>

## Steps to Reproduce
1. Install Node v8
1. Run `node server`/`node upgrade`/`node install`

## Versions
  - Authoring Tool Version: 0.4.0
  - Operating System: Windows 10
  - Node Version: 8.8.1

## Additional Information
I would've liked to do a blanket upgrade of all dependencies but I then got an error with `async` later on in the upgrade script.

Node v8 is due to enter [LTS](https://github.com/nodejs/Release#lts-schedule1) on 31/10/17.

### PR DESCRIPTION
**Pull request by tomgreenfield**
### This PR enables the authoring tool support to Node v8+.
_Any servers running a Node version lower than this must be upgraded._

- To resolve #1769.
- Package dependencies have been updated and syntax fixed where necessary
- Opening this PR for support on how to proceed and to get the ball rolling in prepartion for the refactor
- ~Current hurdle is the loss of `process.domain.session` functionality~
  - ~`process.domain.session` is used to keep a reference to the current request data when we no longer have access to the `req` object~
  - You can debug with Node v8 using Chrome's DevTools running
  ```shellsession
  node --inspect server
  ```
  - ~Breakpoint line [361](https://github.com/adaptlearning/adapt_authoring/blob/49763e9df632dcd4fc7a1ffb295caae631365042/lib/usermanager.js#L361) of lib/usermanager.js to see the issue~

### CODE REVIEW

File: frontend/src/core/app.js [hunk 1]
```diff
@@ -54,6 +55,7 @@
   * Start app load
   */
   loadLibraries(function() {
+    window.Handlebars = $.extend(require('handlebars'), window.Handlebars);
```
Comment (taylortom) [inline frontend/src/core/app.js]: Can this go in `loadLibraries` before the `callback` param is called? (like we do with stuff in `loadCore`)

File: lib/bowermanager.js [hunk 2]
```diff
@@ -2,10 +2,7 @@ var _ = require('underscore');
 var async = require('async');
 var bower = require('bower');
 var fs = require('fs-extra');
-var mkdirp = require('mkdirp');
-var ncp = require('ncp').ncp;
 var path = require('path');
-var rimraf = require('rimraf');
 var semver = require('semver');
 
 var installHelpers = require('./installHelpers');
```
Comment (canstudios-louisem) [inline lib/bowermanager.js]: If we are going to node 8 and dropping 4 could we convert var's like this to `const`. Im not suggesting we do everything but areas as we touch them.

**Comment by canstudios-louisem** [comment]
Should probaly update the .travis.yml to test "8". I can close [this](https://github.com/adaptlearning/adapt_authoring/pull/1689) pr then.

**Comment by taylortom** [comment]
This is all very exciting

**Comment by lc-thomasberger** [comment]
we could also use [async functions](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/await)

**Comment by taylortom** [comment]
I'm getting an issue with Handlebars not being defined when the front-end's been compiled for production
```
Uncaught TypeError: Handlebars.template is not a function
```

**Comment by lc-thomasberger** [comment]
when I do an install on my Windows Machine with node 8.11.2 it fails [here](https://github.com/adaptlearning/adapt_authoring/blob/issue/1769/lib/database.js#L512)
`db.conn.readyState` is equal to 2. 
I am running mongod Version v3.4.9-5-gb2938fa with no auth.

http://mongoosejs.com/docs/api.html#connection_Connection-readyState

**Comment by lc-thomasberger** [comment]
@tomgreenfield thanks, yout fix solved the issue for me. Did a test with importing a course and a plugin. All works good!

**Comment by mike-st** [comment]
Can some one give me an exact Version of node 8 I should be running I just tried 8.11.3 with no luck.

**Comment by lc-thomasberger** [comment]
@mike-st I use 8.11.2 on Windows and so far I have no issues. Do you have more information about your issue? Have you deleted the old node_modules and installed again with the new node version?

**Comment by mike-st** [comment]
Thanks @lc-thomasberger I installed that version like you mentioned and removed the node_modules folder as I forgot the last time. I uninstalled and re-installed that version. Unfortunately it is still not working...

upgrading-node

**Comment by mike-st** [comment]
I have fixed it getto style by changing my original adapt_authoring folder name. installing a fresh adapt authoring tool then coping it's node_modules folder in to my original adapt_authoring folder and deleting the new adapt install.

**Pull request merged**